internal/integration/testcase_test.go

```diff
@@ -39,19 +39,23 @@ type testcase struct {
 	PipelineName string
 }
 
+// k8s labels are limited to length 63, we use the pipeline name as a label.
+// So we sometimes need to limit the length of the pipeline name too.
+func (t *testcase) ShortPipelineName() string {
+	return t.PipelineName[:min(len(t.PipelineName), 63)]
+}
+
 func (t testcase) Init() testcase {
 	t.Helper()
 	t.Parallel()
 
 	if t.PipelineName == "" {
-		queuePrefix := "queue_"
-		namePrefix := "agent-stack-k8s-test-"
-		nameVariable := fmt.Sprintf("%s-%d", strings.ToLower(t.Name()), time.Now().UnixNano())
-		hash := fmt.Sprintf("%x", sha256.Sum256([]byte(nameVariable)))
-
-		// k8s labels are limited to length 63, we use the pipeline name as a label.
-		// So we need to limit the length of the pipeline name too.
-		t.PipelineName = fmt.Sprintf("%s%s", namePrefix, hash[:63-len(namePrefix)-len(queuePrefix)])
+		namePrefix := t.Name()
+		jobID := os.Getenv("BUILDKITE_JOB_ID")
+		if jobID == "" {
+			jobID = strconv.FormatInt(time.Now().Unix(), 10)
```

Comment:
Is that possibly going to cause a problem? Two tests with the same name in the same second with an empty `BUILDKITE_JOB_ID`?

Reply:
Whoops, I should have made it `UnixNano()`. The only time this should be run without a job id is if you're running it locally, in which case you just fix it 🤷‍♂️

Reply:
Its nanoseconds now.